run-bindgen.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+bindgen                                    \
+	--whitelist-type 'nng_.*'              \
+	--whitelist-function 'nng_.*'          \
+	--whitelist-var 'NNG_.*'               \
+	--no-prepend-enum-name                 \
+	--default-enum-style "consts"          \
+	--opaque-type 'nng_pipe_s'             \
+	--opaque-type 'nng_socket_s'           \
+	--opaque-type 'nng_dialer_s'           \
+	--opaque-type 'nng_listener_s'         \
+	--opaque-type 'nng_ctx_s'              \
+	--with-derive-default                  \
+	--use-core                             \
+	--output src/lib.rs                    \
```

Comment:
Rather than output directly to `lib.rs`, might be handy to output to another file.
We can manually maintain lib.rs (adding things like the initializer macros, etc.) and re-export the auto-generated file.  Especially because rust doesn't let you `impl` structs in another module; that's why the `from_i32` helpers were in there.

Reply:
I agree. I'll make the changes soon.